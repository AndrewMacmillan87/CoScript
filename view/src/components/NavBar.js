import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import Button from "./buttons/Button";

import { getUserMessages } from "../redux/auxiliary/user-messages";

// Redux
import { connect } from "react-redux";
import { getMostRecentProject } from "../redux/actions/documents";

import PropTypes from "prop-types";

import "./navbar.css";

export class NavBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            visible: false,
            unread: false,
        };
        this.handleClick = this.handleClick.bind(this);
    }

    /**
     * Checks to see if the user has any unread messages
     * in their personal profile message inbox and sets
     * the 'unread' state property to true if they have.
     * This displays a small icon to the user to indicate
     * they have unread messages
     */
    async componentDidMount() {
        const userMessages = await getUserMessages();
        for (let message of userMessages) {
            if (!message.read) {
                this.setState({
                    unread: true,
                });
            }
        }
    }

    /**
     * Sets the 'visible' state property so that the
     * dropdown menu can be revealed and hidden by
     * clicking the link
     */
    handleClick() {
        this.setState((state) => ({
            visible: !state.visible,
        }));
    }

    render() {
        const { unread } = this.state;
        const { user, loading, history } = this.props;

        const menuIcon = require("../img/menu.svg");
        const spinner = require("../img/loading.svg");
        const isLoggedIn = user && user.isLoggedIn | false;

        const loggedIn = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                        <Link
                            to='#'
                            onClick={async () => {
                                await this.props.getMostRecentProject();
                                const { projectData } = this.props;
                                projectData
                                    ? history.push(`/editor${projectData.url}`)
                                    : history.push("/editor");
                            }}>
                            Workspace
                        </Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    {user && !loading ? (
                        <div className='logged-in-user-container'>
                            <p className='user-username'>
                                {user.profile.displayName}
                            </p>
                            <button
                                className='desktop-icon-link'
                                onClick={this.handleClick}>
                                {unread ? (
                                    <div className='avatar-unread-msg-container'>
                                        <div className='avatar-unread-msg-icon'></div>
                                        <img
                                            className='user-gravatar'
                                            src={user.avatar}
                                            alt='user gravatar'
                                        />
                                    </div>
                                ) : (
                                    <img
                                        className='user-gravatar'
                                        src={user.avatar}
                                        alt='user gravatar'
                                    />
                                )}
                            </button>
                        </div>
                    ) : (
                        <img
                            className='user-spinner'
                            src={spinner}
                            alt='loading spinner icon'
                        />
                    )}
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                    <Link
                        to='#'
                        onClick={async () => {
                            await this.props.getMostRecentProject();
                            const { projectData } = this.props;
                            projectData
                                ? history.push(`/editor${projectData.url}`)
                                : history.push("/editor");
                        }}>
                        <p className='dropdown-link'>Workspace</p>
                    </Link>
                    <Link to='/profile'>
                        <p className='dropdown-link'>My Profile</p>
                    </Link>
                    <a
                        onClick={() => {
                            history["from"] = "logout";
                            history.push("/redirect");
                        }}
                        href=' '>
                        <p className='dropdown-link'>Logout</p>
                    </a>
                </div>
            </Fragment>
        );

        const visiting = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    <div>
                        <Link to='/register'>
                            <Button classes='btn' value='Register' />
                        </Link>
                    </div>
                    <div>
                        <Link to='/login'>
                            <Button classes='grn-btn' value='Login' />
                        </Link>
                    </div>
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/register'>
                        <p className='dropdown-link'>Register</p>
                    </Link>
                    <Link to='/login'>
                        <p className='dropdown-link'>Login</p>
                    </Link>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                </div>
            </Fragment>
        );

        return (
            <nav className='navbar'>
                {!loading && (
                    <Fragment>{!isLoggedIn ? visiting : loggedIn}</Fragment>
                )}
            </nav>
        );
    }
}

NavBar.propTypes = {
    user: PropTypes.object,
    loading: PropTypes.bool,
    getMostRecentProject: PropTypes.func.isRequired,
};

const setProps = (state) => ({
    user: state.auth.user,
    loading: state.auth.loading,
    projectData: state.documents.projectData,
});

export default connect(setProps, {
    getMostRecentProject,
})(withRouter(NavBar));
