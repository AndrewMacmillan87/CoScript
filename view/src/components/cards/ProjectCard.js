import React from "react";

import Button from "../buttons/Button";

import PropTypes from "prop-types";

import "./project-card.css";

const ProjectCard = ({
    src,
    img_alt,
    title,
    project_admin,
    genres,
    published,
    onClick
}) => {
    return (
        <div className='card-item'>
            <div className='card-box-image-container'>
                <img className='card-box-image' src={src} alt={img_alt} />
            </div>
            <div className='card-item-details'>
                <p>{title}</p>
                <p>{project_admin}</p>
                <p className='genres'>{genres}</p>

                <Button
                    classes='small-btn'
                    value={published ? "Read" : "Details"}
                    onClick={onClick}
                />
            </div>
        </div>
    );
};

ProjectCard.propTypes = {
    src: PropTypes.string,
    img_alt: PropTypes.string,
    title: PropTypes.string,
    project_admin: PropTypes.string,
    genres: PropTypes.string
};

export default ProjectCard;
