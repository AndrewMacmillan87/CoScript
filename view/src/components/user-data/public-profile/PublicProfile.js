import React, { Fragment } from "react";
import { withRouter } from "react-router-dom";

// Redux
import { connect } from "react-redux";
import { loadProject } from "../../../redux/actions/documents";
import { sendFriendRequest, loadUserProfile } from "../../../redux/actions/profile";

import ProjectCard from "../../cards/ProjectCard";
import NavBar from "../../NavBar";
import Footer from "../../Footer";
import Alert from "../../alert/Alert";

import PropTypes from "prop-types";

import "./public-profile.css";

export class PublicProfile extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loading: true,
            revealRequestWindow: false,
            submitted: false,
        };
        this.showRequestWindow = this.showRequestWindow.bind(this);
        this.removeRequestWindow = this.removeRequestWindow.bind(this);
        this.showSubmitSpinner = this.showSubmitSpinner.bind(this);
        this.removeSubmitSpinner = this.removeSubmitSpinner.bind(this);
        this.sendFriendRequest = this.sendFriendRequest.bind(this);
    }

    /**
     * Sets the browser tab title and scrolls to the
     * top of the page when the component mounts. Also
     * loads the users personal profile into the
     * frontend
     */
    async componentDidMount() {
        document.title = "CoScript - Public Profile";
        window.scrollTo(0, 0);
        const { userProfile, loadUserProfile } = this.props;
        if (userProfile) {
            await loadUserProfile(userProfile.id);
        }
        setTimeout(() => {
            this.setState({
                loading: false,
            });
        }, 500);
    }

    /**
     * Sets the 'revealRequestWindow' to true to
     * display a window to the user asking them if
     * they want to send a friend request to the
     * owner of the profile
     */
    showRequestWindow() {
        const { user } = this.props;
        if (user) {
            this.setState({
                revealRequestWindow: true,
            });
        }
    }

    /**
     * Sets the 'revealRequestWindow' to false to
     * remove a window tasking a user if they
     * want to send a friend request to the owner
     * of the profile
     */
    removeRequestWindow() {
        this.setState({
            revealRequestWindow: false,
        });
    }

    /**
     * Sets the 'submitted' state property to true
     * to display a loading spinner while the user's
     * profile loads
     */
    showSubmitSpinner() {
        this.setState({
            submitted: true,
        });
    }

    /**
     * Sets the 'submitted' state property to false
     * to remove a loading spinner once the user's
     * profile has loaded
     */
    removeSubmitSpinner() {
        this.setState({
            submitted: false,
        });
    }

    /**
     * Sends a friend request message to the personal
     * profile message inbox of the user who owns this
     * public profile.
     *
     * @param {Object} e - event object
     */
    async sendFriendRequest(e) {
        e.preventDefault();
        this.showSubmitSpinner();
        const { user, userProfile, sendFriendRequest } = this.props;
        const msgText = `${user.profile.displayName} sent you a friend request`;
        const success = await sendFriendRequest(msgText, user.userId, userProfile.id);
        if (success) {
            this.setState({
                revealRequestWindow: false,
            });
            this.removeSubmitSpinner();
        } else {
            setTimeout(() => {
                this.removeRequestWindow();
            }, 1000);
        }
    }

    render() {
        const { loading, revealRequestWindow, submitted } = this.state;
        const { user, userProfile, loadProject, history } = this.props;

        const spinner = require("../../../img/loading-black.svg");
        const sky = require("../../../img/sky-example.jpg");

        return (
            <Fragment>
                <Alert />
                <NavBar />
                {revealRequestWindow && (
                    <div className='collaborate-box'>
                        {" "}
                        <form onSubmit={(e) => this.sendFriendRequest(e)}>
                            <p className='collaborate-heading'>
                                Send <b>{userProfile && userProfile.name}</b> a friend request?
                            </p>
                            <div className='btn-container'>
                                {submitted && <img className='reply-msg-loading-spinner' src={spinner} alt='loading' />}
                                <button className='cancel-btn' onClick={this.removeRequestWindow}>
                                    Cancel
                                </button>
                                <input className='confirm-btn' type='submit' value='Send' />
                            </div>
                        </form>
                    </div>
                )}
                <main className='public-profile-main'>
                    {loading ? (
                        <div className='public-profile-loading-spinner-container'>
                            <img className='public-profile-loading-spinner' src={spinner} alt='loading' />
                        </div>
                    ) : (
                        <Fragment>
                            <header className='public-profile-header'>
                                <img className='public-profile-avatar' src={userProfile && userProfile.avatar} alt='user-avatar' />
                                <div className='profile-contact-form'>
                                    <p className='public-profile-display-name'>{userProfile && userProfile.name} </p>
                                    {user && userProfile && user.userId !== userProfile.id && (
                                        <button
                                            className='profile-connect-btn'
                                            onClick={
                                                user
                                                    ? this.showRequestWindow
                                                    : () => {
                                                          history.push("/login");
                                                      }
                                            }>
                                            <span className='profile-pipe'>|</span> Connect
                                        </button>
                                    )}
                                </div>
                            </header>
                            <section className='public-profile-info-main'>
                                <div className='public-profile-info-container'>
                                    <div className='public-profile-info about'>
                                        <h2 className='public-profile-heading'>About</h2>
                                        <div className='public-profile-about '>
                                            {userProfile && (
                                                <Fragment>
                                                    {userProfile.about
                                                        ? userProfile.about
                                                        : `${userProfile.name} has not added any personal information yet`}
                                                </Fragment>
                                            )}
                                        </div>
                                    </div>
                                    <div className='public-profile-info authors'>
                                        <h2 className='public-profile-heading'>Favourite Authors</h2>
                                        <div>
                                            {userProfile && userProfile.authors.length
                                                ? userProfile.authors.map((author) => {
                                                      return <p key={author}>{author}</p>;
                                                  })
                                                : "None added yet"}
                                        </div>
                                    </div>
                                    <div className='public-profile-info books'>
                                        <h2 className='public-profile-heading'>Favourite Books</h2>
                                        <div>
                                            {userProfile && userProfile.books.length
                                                ? userProfile.books.map((book) => {
                                                      return <p key={book}>{book}</p>;
                                                  })
                                                : "None added yet"}
                                        </div>
                                    </div>
                                </div>
                            </section>
                            <section className='public-profile-user-projects'>
                                <h2 className='public-profile-heading projects'>Projects</h2>
                                <div className='user-project-cards'>
                                    {userProfile &&
                                        userProfile.projects.map((project) => {
                                            return (
                                                <ProjectCard
                                                    key={project.projectId}
                                                    src={sky}
                                                    img_alt='Project cover image'
                                                    title={project.title}
                                                    project_admin={project.author}
                                                    genres={project.genres}
                                                    projectUrl={project.url}
                                                    onClick={async () => {
                                                        await loadProject(project);
                                                        history.push(`/landing${project.url}`);
                                                    }}
                                                />
                                            );
                                        })}
                                </div>
                            </section>
                        </Fragment>
                    )}
                </main>
                <Footer />
            </Fragment>
        );
    }
}

PublicProfile.propTypes = {
    user: PropTypes.object,
    userProfile: PropTypes.object,
    loadProject: PropTypes.func.isRequired,
    sendFriendRequest: PropTypes.func.isRequired,
    loadUserProfile: PropTypes.func.isRequired,
};

const mapProps = (state) => ({
    user: state.auth.user,
    userProfile: state.auth.userProfile,
});

export default connect(mapProps, {
    loadUserProfile,
    loadProject,
    sendFriendRequest,
})(withRouter(PublicProfile));
