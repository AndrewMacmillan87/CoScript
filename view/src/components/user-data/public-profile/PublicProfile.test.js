import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../../redux/store/reduxStore";

import { PublicProfile } from "./PublicProfile";
import NavBar from "../../NavBar";
import ProjectCard from "../../cards/ProjectCard";

import projectArray from "../../../redux/tests/auxiliary/data/projects.json";

describe("PublicProfile component", () => {
    let props, mockLoadProject, mockSendFriendRequest, mockLoadUserProfile;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        mockSendFriendRequest = jest.fn();
        mockLoadUserProfile = jest.fn();
        props = {
            user: { userId: "654321", profile: { displayName: "Test Name" } },
            userProfile: { avatar: "", name: "Test Name", id: "123456", about: "Test about", authors: [], books: [], projects: null },
            loadProject: jest.fn(),
            sendFriendRequest: mockSendFriendRequest,
            loadUserProfile: mockLoadUserProfile,
            history: {
                from: "",
                push: jest.fn(),
            },
        };
    });

    afterEach(() => {
        props.userProfile.projects = null;
    });

    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <PublicProfile {...props} />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        global.scrollTo = jest.fn();
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should call `loadUserProfile()` on mount", () => {
        mount(
            <Provider store={reduxStore}>
                <Router>
                    <PublicProfile {...props} />
                </Router>
            </Provider>
        );
        expect(mockLoadUserProfile).toBeCalled();
    });

    it("should render the NavBar component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <PublicProfile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(NavBar).length).toEqual(1);
    });

    it("should render a main element with a className of 'public-profile-main'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <PublicProfile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("main").length).toEqual(1);
        expect(wrapper.find(".public-profile-main").length).toEqual(1);
    });

    it("should render an img element with a className of 'public-profile-loading-spinner' if 'loading is true'", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <PublicProfile {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("img").length).toEqual(1);
        expect(wrapper.find(".public-profile-loading-spinner").length).toEqual(1);
    });

    describe("'loading is false", () => {
        beforeEach(() => {
            props.userProfile.projects = projectArray.projects;
        });

        it("should render a header element", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find("header").length).toEqual(1);
        });

        it("should render a button element with a className of 'profile-connect-btn'", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find("button").length).toEqual(7);
            expect(wrapper.find(".profile-connect-btn").length).toEqual(1);
        });

        it("should render 7 span element", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find("span").length).toEqual(7);
        });

        it("should render 2 section elements", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find("section").length).toEqual(2);
        });

        it("should render 33 div elements", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find("div").length).toEqual(33);
        });

        it("should render 5 h2 elements", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find("h2").length).toEqual(5);
        });

        it("should render 6 projectCard components", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find(ProjectCard).length).toEqual(6);
        });

        it("should render 6 projectCard components", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ loading: false });
            expect(wrapper.find(ProjectCard).length).toEqual(6);
        });
    });

    describe("revealRequestWindow is true", () => {
        it("should render a div element with a className of collaborate-box", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );

            wrapper.find(PublicProfile).setState({ revealRequestWindow: true });
            expect(wrapper.find(".collaborate-box").length).toEqual(1);
        });

        it("should render a form element", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );

            wrapper.find(PublicProfile).setState({ revealRequestWindow: true });
            expect(wrapper.find("form").length).toEqual(1);
        });

        it("should render a button element", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ revealRequestWindow: true });
            expect(wrapper.find("button").length).toEqual(1);
        });

        it("should render an input element with a className of 'confirm-btn'", () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ revealRequestWindow: true });
            expect(wrapper.find(".confirm-btn").length).toEqual(1);
        });

        it("should call `sendFriendRequest()` when the form is submitted", async () => {
            const wrapper = mount(
                <Provider store={reduxStore}>
                    <Router>
                        <PublicProfile {...props} />
                    </Router>
                </Provider>
            );
            wrapper.find(PublicProfile).setState({ revealRequestWindow: true });
            wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
            await expect(mockSendFriendRequest).toBeCalled();
        });
    });
});
