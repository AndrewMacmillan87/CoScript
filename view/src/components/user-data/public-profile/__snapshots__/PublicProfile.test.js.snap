// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicProfile component renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  />,
  <main
    className="public-profile-main"
  >
    <div
      className="public-profile-loading-spinner-container"
    >
      <img
        alt="loading"
        className="public-profile-loading-spinner"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "loading-black.svg",
          }
        }
      />
    </div>
  </main>,
  <footer
    className="footer"
  >
    <h2
      className="footer-h2"
    >
      CoScript
    </h2>
    <div
      className="footer-container"
    >
      <div
        className="links-container socials"
      >
        <p>
          Socials
        </p>
        <div
          className="footer-links"
        >
          <a
            href="https://github.com/AndrewMacmillan87/coscript"
          >
            GitHub
          </a>
          <a
            href="/"
          >
            Twitter
          </a>
          <a
            href="/"
          >
            Facebook
          </a>
        </div>
      </div>
      <div
        className="links-container site-map"
      >
        <p>
          Site Map
        </p>
      </div>
      <div
        className="links-container legals"
      >
        <p>
          Legal
        </p>
        <div
          className="footer-links"
        >
          <a
            href="/terms"
          >
            Terms and Conditions
          </a>
          <a
            href="/copyright"
          >
            Copyright Info
          </a>
          <a
            href="/agreement"
          >
            Writers Agreements
          </a>
          <a
            href="/"
          >
            Accreditation
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
